lib/smart_proxy_openscap/policy_guide.rb

```diff
@@ -6,21 +6,20 @@ class PolicyGuide
       include ::Proxy::Log
 
       def generate_guide(file_content, policy_id)
-        file = Tempfile.new
-        file.write file_content
-        file.rewind
-        `oscap xccdf generate #{profile_opt policy_id} guide #{file.path}`
+        Tempfile.create do |file|
+          file.write file_content
+          file.flush
+          command = ['oscap', 'xccdf', 'generate', profile_opt(policy_id), 'guide', file.path].flatten
```

Comment:
I like avoiding `.flatten` when I can.
```suggestion
          command = ['oscap', 'xccdf', 'generate'] + profile_opt(policy_id) + ['guide', file.path]
```

And if the command supports it, this is easier to read
```suggestion
          command = ['oscap', 'xccdf', 'generate', 'guide', file.path] + profile_opt(policy_id)
```

Taking it a step further, I'd then also consider dropping the helper:
```suggestion
          command = ['oscap', 'xccdf', 'generate', 'guide', file.path]
          command += ['--profile', policy_id] if policy_id
```

Reply:
It seems that encountering first non-option and non-subcommand argument terminates the option parsing, so `--profile $value` cannot be at the very end. I'll go with the middle option.

> I like avoiding .flatten when I can.

Is that just personal preference or is there a reason why you'd advise to avoid it in general?

Reply:
> It seems that encountering first non-option and non-subcommand argument terminates the option parsing, so `--profile $value` cannot be at the very end. I'll go with the middle option.

I was afraid of that. I suspect that `--profile` is an option to `generate` and `guide` is another subcommand, but I'm just guessing about its internal implementation. I have some vague recollection that Python's `argparse` works that way.

> Is that just personal preference or is there a reason why you'd advise to avoid it in general?

I think more of a personal preference. In my mind I like to be able to reason about data types and without flatten it's `Array[String]` but with it's sort of `Array[Variant[String, Array]]` but the `Array` can't be fully specified since it's an unknown depth. There could be a performance difference, but I think that's not really why I avoid it and probably completely irrelevant here given executing a command is going to be way more expensive compared to `flatten`.